src/Angular.js

```diff
@@ -978,6 +978,20 @@ function equals(o1, o2) {
   return false;
 }
 
+var unsafeEval = function() { /* jshint ignore:line */
+  if (isDefined(unsafeEval.isActive_)) return unsafeEval.isActive_;
+
+  var active = true;
+  try {
+    /* jshint -W031, -W054 */
+    new Function('');
+    /* jshint +W031, +W054 */
+  } catch (e) {
+    active = false;
+  }
+  return (unsafeEval.isActive_ = active);
+};
```

Comment:
Having `unsafeEval` split out of `csp` may make it chatty in the console. I can add guards to it if you all would like.